### DISCUSSION TITLE
Restrict kql search query to specific div in a page

### DISCUSSION THREAD

**Discussion by webbrewer** [microsoft-search/pnp-modern-search#2279] [General]
I'm using v3 PnP modern search web parts and restricting my search to published news posts but on the new post pages, there's a "recent posts" news web part so search includes content from that web part. As an example I have posts titled "post a" and "post b". If I search for "a"  I get both posts returned because search (correctly) finds the "b" in the news web part.   Is there a way to refine the search query to only look at the "canvascontent" field ie the body  of the post, excluding other web parts on the page? Something like:

`{searchTerms} Path: "https://xxx.sharepoint.com/sites/xxx/SitePages" RefinableString01:xxx PromotedState:2 *FIELD:CANVASCONTENT*

**Comment by FranckyC** [comment]
Hi @webbrewer, did you try to map the `ows_CanvasContent`1 to a `RefinableString` property and use it in your query?
![image](https://user-images.githubusercontent.com/5659122/184541815-44840267-16f2-4897-a330-a17fe5115c8b.png)

**Reply by wobba** [reply]
You can also try to use `CanvasContent1OWSHTML:(your terms)`. That said, searchable content of a web part is already indexed, if the web part was set for the content to be indexed. I'm guessing you have many metadata fields you don't want to match against?

**Reply by webbrewer** [reply]
@FranckyC @wobba Yes, thank you I found this  suggestion somewhere else, and it worked: 
`contentclass:STS_ListItem_WebPageLibrary IsContainer<>true RefinableString09:{searchTerms} -ContentClass=urn:content-class:SPSPeople `
Where ows_CanvasContent1 is mapped Ref String 09
